runtime/lua/vim/lsp/inlay_hint.lua

```diff
@@ -85,6 +85,33 @@ function M.on_inlayhint(err, result, ctx)
   api.nvim__redraw({ buf = bufnr, valid = true, flush = false })
 end
 
+--- Refresh inlay hints, only if we have attached clients that support it
+---@param bufnr (integer) Buffer handle, or 0 for current
+---@param client_id? (integer) Client ID, or nil for all
+---@private
+local function _refresh(bufnr, client_id)
+  for _, client in
+    ipairs(vim.lsp.get_clients({
+      bufnr = bufnr,
+      id = client_id,
+      method = ms.textDocument_inlayHint,
+    }))
+  do
+    for _, winid in ipairs(api.nvim_list_bufs()) do
+      if api.nvim_win_get_buf(winid) == bufnr then
+        local first = vim.fn.line('w0', winid)
+        local last = vim.fn.line('w$', winid)
+        ---@type lsp.InlayHintParams
+        local params = {
+          textDocument = util.make_text_document_params(bufnr),
+          range = util._make_line_range_params(bufnr, first - 1, last - 1, client.offset_encoding),
+        }
+        client:request(ms.textDocument_inlayHint, params, M.on_inlayhint, bufnr)
```

Comment:
Might be a good idea to add a filter option to `vim.lsp.buf_request_all` that's passed to the call of `vim.lsp.get_clients`. WDYT @lewis6991?

Reply:
I'm personally not a fan of `buf_request_all()`.

Reply:
Gotcha. The nice thing about `buf_request_all` is that it handles the request cancellation which will be needed in https://github.com/neovim/neovim/issues/31206, but not a blocker.